src/Planner/Planner.cpp

```diff
@@ -858,9 +858,8 @@ void addWithFillStepIfNeeded(QueryPlan & query_plan,
     query_plan.addStep(std::move(filling_step));
 }
 
-void addLimitByStep(QueryPlan & query_plan,
-    const LimitByAnalysisResult & limit_by_analysis_result,
-    const QueryNode & query_node)
+void addLimitByStep(
+    QueryPlan & query_plan, const LimitByAnalysisResult & limit_by_analysis_result, const QueryNode & query_node, bool do_not_skip_offset)
```

Comment:
Boolean flags with included negation in the name are not a nice thing to think about. 
```suggestion
    QueryPlan & query_plan, const LimitByAnalysisResult & limit_by_analysis_result, const QueryNode & query_node, bool do_skip_offset)
```